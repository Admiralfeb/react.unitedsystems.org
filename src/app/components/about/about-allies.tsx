import {
  List,
  ListItem,
  ListItemText,
  makeStyles,
  Typography,
} from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles({
  header: {
    textAlign: 'center',
  },
  allies: {
    border: '3px solid white',
    borderRadius: 10,
    margin: 5,
  },
});
export const AboutAllies = () => {
  const classes = useStyles();
  return (
    <div>
      <Typography variant='h2' className={classes.header}>
        Allies
      </Typography>
      <div className={classes.allies}>
        <List>
          {allies.map((rule: string, i: number) => (
            <ListItem key={i}>
              <ListItemText primary={`${rule}`} />
            </ListItem>
          ))}
        </List>
      </div>
    </div>
  );
};

const allies = [
  '1st Imperium Auxiliary Cohort',
  'Alba Rotam Private Military',
  'Arbor Caelum',
  'Black Universe Gazers',
  'Daily Routine Incorperated',
  'Explorers of Nabudis',
  'The Galactic Archive',
  'Gilgamesh Corps',
  'Independence Separatist Fleet',
  'Intergalactic Astronomical Union',
  'Interstellar Incorporated',
  'Knights of Pleiades',
  'Ryders of the Void',
  'ROOKS',
  'Sapere Aude',
  'The Altruists',
];
