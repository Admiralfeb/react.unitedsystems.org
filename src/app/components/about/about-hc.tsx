import {
  List,
  ListItem,
  ListItemText,
  makeStyles,
  Typography,
} from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles({
  header: {
    textAlign: 'center',
  },
  rank: {
    border: '3px solid white',
    borderRadius: 10,
    margin: 5,
  },
});
export const AboutHC = () => {
  const classes = useStyles();
  return (
    <div>
      <Typography variant='h2' className={classes.header}>
        High Command
      </Typography>
      <div className={classes.rank}>
        <List>
          {hc.map((rule: string, i: number) => (
            <ListItem key={i}>
              <ListItemText primary={`${rule}`} />
            </ListItem>
          ))}
        </List>
      </div>
    </div>
  );
};

const hc = [
  'NickG2002',
  'Goliath the Second',
  'LOKioo42',
  'Pablok',
  'PracticalHalo',
  'RedFang Culph',
  'UltraNeros',
  'Admiralfeb',
  'Akhaten',
  'Aurelius Publius',
  'Captain Smokeq',
  'Clever Ape',
  'IM2D',
  'Nullyti',
  'Yuting9',
];
