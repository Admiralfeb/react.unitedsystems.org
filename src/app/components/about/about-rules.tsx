import {
  Typography,
  List,
  ListItem,
  ListItemText,
  makeStyles,
} from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles({
  root: {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr',
    gridTemplateRows: 'auto',
  },
  rules: {
    border: '3px solid white',
    borderRadius: 10,
    margin: 5,
  },
  consequences: {
    gridColumn: '1 / 3',
    border: '3px solid white',
    borderRadius: 10,
    margin: 5,
  },
  '@media (max-width: 1024px)': {
    root: {
      gridTemplateColumns: '1fr',
      gridTemplateRows: 'auto',
    },
    consequences: {
      gridColumn: 1,
    },
  },
});

export const AboutRules = () => {
  const classes = useStyles();
  return (
    <div>
      <Typography variant='h3'>Rules</Typography>
      <div className={classes.root}>
        <div className={classes.rules}>
          <Typography variant='h4'>Discord</Typography>
          <List>
            {discordRules.map((rule: string, index: number) => (
              <ListItem key={index}>
                <ListItemText primary={`${index + 1}) ${rule}`} />
              </ListItem>
            ))}
          </List>
        </div>
        <div className={classes.rules}>
          <Typography variant='h4'>Members of USC</Typography>
          <List>
            {memberRules.map((rule: string, index: number) => (
              <ListItem key={index}>
                <ListItemText primary={`${index + 1}) ${rule}`} />
              </ListItem>
            ))}
          </List>
        </div>
        <div className={classes.consequences}>
          <Typography>
            Those found in violation of any of these Discord or Group rules, or
            found attempting to bypass these rules in any way, will first be
            subject to a minimum of a warning, further offences will be subject
            to demotion, expulsion, and/or USC Kill-on-Sight status.
          </Typography>
        </div>
      </div>
    </div>
  );
};

const discordRules = [
  'Be Respectful',
  'Be Mature',
  'Do not @everyone',
  'No porn or nudity',
  'No harassment',
  'No excessive swearing',
  'No derogatory terms or phrases',
  'No political discussions as they can become heated.',
];

const memberRules = [
  'Will not, under no circumstance, Combat Log.',
  'Will not attack other USC Personnel or any allies of the United Systems Cooperative.',
  'Will not attack any innocent CMDRs, unless in pre-arranged combat, or for Powerplay.',
  'Will not, under no circumstance, discriminate against other commanders on the basis of race, religion, sexuality, nationality, etc.',
  'Will not, under no circumstance, seek revenge against other USC Personnel, directly or indirectly. Punishments for crimes, and resolutions for transgressions, will be handed out by High Command.',
  'Will not, under no circumstance, to divulge information marked as confidential by High Command.',
];
