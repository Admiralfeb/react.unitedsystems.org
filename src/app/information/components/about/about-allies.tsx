import {
  List,
  ListItem,
  ListItemText,
  makeStyles,
  Paper,
  Typography,
} from '@material-ui/core';
import React from 'react';

const useStyles = makeStyles({
  header: {
    textAlign: 'center',
  },
  allies: {
    border: '3px solid white',
    borderRadius: 10,
    margin: 'auto',
    width: 450,
  },
});
export const AboutAllies = () => {
  const classes = useStyles();
  return (
    <div>
      <Typography variant='h3' className={classes.header}>
        Allies
      </Typography>
      <Paper className={classes.allies}>
        <List>
          {allies.map((rule: string, i: number) => (
            <ListItem key={i}>
              <ListItemText primary={`${rule}`} />
            </ListItem>
          ))}
        </List>
      </Paper>
    </div>
  );
};

const allies = [
  '1st Imperium Auxiliary Cohort',
  'Alba Rotam Private Military',
  'Arbor Caelum',
  'Black Universe Gazers',
  'Daily Routine Incorperated',
  'Explorers of Nabudis',
  'The Galactic Archive',
  'Gilgamesh Corps',
  'Independence Separatist Fleet',
  'Intergalactic Astronomical Union',
  'Interstellar Incorporated',
  'Knights of Pleiades',
  'Ryders of the Void',
  'ROOKS',
  'Sapere Aude',
  'The Altruists',
];
