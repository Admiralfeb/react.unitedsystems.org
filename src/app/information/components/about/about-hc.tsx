import {
  List,
  ListItem,
  ListItemText,
  makeStyles,
  Paper,
  Typography,
} from '@material-ui/core';

const useStyles = makeStyles({
  root: {},
  header: {
    textAlign: 'center',
  },
  rank: {
    border: '3px solid white',
    borderRadius: 10,
    margin: 'auto',
    width: 450,
  },
});
export const AboutHC = () => {
  const classes = useStyles();
  return (
    <div>
      <Typography variant='h3' className={classes.header}>
        High Command
      </Typography>
      <Paper className={classes.rank}>
        <List>
          {hc.map((rule: string, i: number) => (
            <ListItem key={i}>
              <ListItemText primary={`${rule}`} />
            </ListItem>
          ))}
        </List>
      </Paper>
    </div>
  );
};

const hc = [
  'NickG2002',
  'Goliath the Second',
  'LOKioo42',
  'Pablok',
  'PracticalHalo',
  'RedFang Culph',
  'UltraNeros',
  'Admiralfeb',
  'Akhaten',
  'Aurelius Publius',
  'Captain Smokeq',
  'Clever Ape',
  'IM2D',
  'Nullyti',
  'Queen Eleanor',
  'Yuting9',
];
