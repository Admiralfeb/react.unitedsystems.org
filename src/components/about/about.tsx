import { makeStyles, Typography } from '@material-ui/core';
import { useInfoButtons } from 'hooks/useInfoButtons';
import { AboutLinks } from './aboutLinks';

const useStyles = makeStyles((theme) => ({
  header: {
    textAlign: 'center',
  },
  hcAndAllies: {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr',
  },
}));

export const About = () => {
  const classes = useStyles();
  const { uscLinksList } = useInfoButtons();
  return (
    <>
      <Typography variant="h3" className={classes.header}>
        About USC
      </Typography>
      <AboutLinks
        id="usc-links"
        key="usc-links"
        header="USC Links"
        buttons={uscLinksList}
      />
    </>
  );
};
