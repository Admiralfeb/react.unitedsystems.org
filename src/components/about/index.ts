export * from './aboutAllies';
export * from './aboutHC';
export * from './aboutRules';
export * from './carriers';
export * from './about';
