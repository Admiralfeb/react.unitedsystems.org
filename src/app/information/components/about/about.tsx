import { makeStyles, Paper, Typography } from '@material-ui/core';

import { AboutAllies } from './about-allies';
import { AboutHC } from './about-hc';
import { AboutRules } from './about-rules';

const useStyles = makeStyles({
  root: {
    height: '100%',
  },
  header: {
    textAlign: 'center',
  },
  paper: {
    textAlign: 'center',
    width: '80%',
    margin: 'auto',
    padding: 5,
    paddingBottom: 10,
    marginBottom: 5,
  },
  rules: {
    border: '3px solid white',
    borderRadius: 10,
    margin: 5,
  },
  setGrid: {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr',
  },
  '@media (max-width: 1024px)': {
    setGrid: {
      gridTemplateColumns: '1fr',
      gridTemplateRows: '1fr 1fr',
    },
  },
});

export const About = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Typography variant="h2" className={classes.header}>
        About USC
      </Typography>
      <Paper className={classes.paper}>
        <AboutRules />
        <div className={classes.setGrid}>
          <AboutHC />
          <AboutAllies />
        </div>
      </Paper>
    </div>
  );
};
