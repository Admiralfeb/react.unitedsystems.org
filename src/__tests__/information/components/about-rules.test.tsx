import { cleanup, render, RenderResult } from '@testing-library/react';
import { AboutRules } from '../../../app/information/components/about/about-rules';

let componentBody: RenderResult;

describe('About Rules', () => {
  beforeEach(() => {
    componentBody = render(<AboutRules />);
  });

  afterEach(() => {
    cleanup();
  });

  it(`should match the snapshot`, () => {
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
