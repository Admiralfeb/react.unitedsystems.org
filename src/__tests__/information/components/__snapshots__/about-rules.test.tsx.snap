// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Rules should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-header-1"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Rules
      </h3>
      <div
        class="MuiPaper-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="makeStyles-rules-3"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Discord
          </h4>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  1) Be Respectful
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  2) Be Mature
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  3) Do not @everyone
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  4) No porn or nudity
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  5) No harassment
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  6) No excessive swearing
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  7) No derogatory terms or phrases
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  8) No political discussions as they can become heated.
                </span>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="makeStyles-rules-3"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Members of USC
          </h4>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  1) Will not, under no circumstance, Combat Log.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  2) Will not attack other USC Personnel or any allies of the United Systems Cooperative.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  3) Will not attack any innocent CMDRs, unless in pre-arranged combat, or for Powerplay.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  4) Will not, under no circumstance, discriminate against other commanders on the basis of race, religion, sexuality, nationality, etc.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  5) Will not, under no circumstance, seek revenge against other USC Personnel, directly or indirectly. Punishments for crimes, and resolutions for transgressions, will be handed out by High Command.
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  6) Will not, under no circumstance, to divulge information marked as confidential by High Command.
                </span>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="makeStyles-consequences-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Those found in violation of any of these Discord or Group rules, or found attempting to bypass these rules in any way, will first be subject to a minimum of a warning, further offences will be subject to demotion, expulsion, and/or USC Kill-on-Sight status.
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
